### PR TITLE
Pull Requests: Add merge endpoint

### LINKED ISSUE #8227: Pull Requests MVP: Add pull request merge endpoint
Add an endpoint to merge pull request so that the merge operation and updated of pull request status will be performed in side the server itself.

At first we thought this should be out of scope for MVP - but it just complicates things instead of simplifying

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #8227 

## Change Description

Add an endpoint for pulls which will handle merge of pull requests

### Testing Details

Added controller unit tests

### CODE REVIEW

File: pkg/api/controller.go [hunk 1]
```diff
@@ -5314,6 +5315,72 @@ func (c *Controller) UpdatePullRequest(w http.ResponseWriter, r *http.Request, b
...
+	}
+
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, "missing user")
+		return
+	}
+
+	// Attempt to merge branches
+	reference, err := c.Catalog.Merge(ctx, repository, pr.Destination, pr.Source, user.Committer(), "", nil, "")
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/api/controller.go:5358]: Is it needed in the listing page?
If not, it should be under the `Read` permission data and not provided under listing

File: pkg/graveler/graveler.go [hunk 2]
```diff
@@ -292,9 +292,10 @@ type PullRequestRecord struct {
 }
 
 type UpdatePullRequest struct {
-	Title       *string
-	Description *string
-	Status      *string
+	Title          *string
+	Description    *string
+	Status         *string
+	MergedCommitID *string
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/graveler/graveler.go:298]: True

File: pkg/api/controller.go [hunk 3]
```diff
@@ -5213,6 +5213,7 @@ func (c *Controller) ListPullRequests(w http.ResponseWriter, r *http.Request, re
 			DestinationBranch: p.DestinationBranch,
 			SourceBranch:      p.SourceBranch,
 			MergedCommitId:    nil,
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/api/controller.go:5215]: Is it needed in the listing page?
If not, it should be under the Read permission data and not provided under listing

File: pkg/catalog/catalog.go [hunk 4]
```diff
@@ -3005,6 +3005,7 @@ func (c *Catalog) ListPullRequest(ctx context.Context, repositoryID, prefix stri
 			SourceBranch:      v.Source,
 			DestinationBranch: v.Destination,
 			CreationDate:      v.CreationDate,
+			ClosedDate:        v.ClosedDate,
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/catalog/catalog.go:3008]: Is it needed in the listing page?
If not, it should be under the Read permission data and not provided under listing

File: pkg/api/controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/api/controller.go:5358]: Better approach is to add metadata

[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/api/controller.go:5358]: The merge commit message by default (when passing empty) is: `Merge 'X' into 'Y'`

File: pkg/api/controller.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (itaigilo) [reply pkg/api/controller.go:5215]: It's not needed at the moment. It's just seems weird to me to leave it empty if we already have the data.

File: pkg/api/controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (itaigilo) [reply pkg/api/controller.go:5358]: Oh, so the default is good enough.

File: pkg/catalog/catalog.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (itaigilo) [reply pkg/catalog/catalog.go:3008]: See previous comment.

File: pkg/api/controller.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/api/controller.go:5215]: Same reason why we leave the description empty - permissions management

**Pull request merged**